### PR TITLE
add: recalculate checksum after link creation, test

### LINKED ISSUE #1221: state: add entry without manual checksum recalculation
Currently when `dvc add` runs, it doesn't add state entry describing a linked file, leaving it for the next operation to manually recompute. Need to add state entry right after we've created links.

### PR DESCRIPTION
**Pull request by pared**
Fixes #1221

### CODE REVIEW

File: dvc/remote/local.py [hunk 1]
```diff
@@ -385,6 +385,10 @@ def _save_file(self, path_info):
         self.link(cache, path)
         self.state.update_link(path)
 
+        # md5 is known, write method to update for known path and md5
+        # self.state.update_state_for_known_path_checksum(path, md5)
+        self.state.update(path, md5)
```
Comment (efiop) [inline dvc/remote/local.py]: Let's also `self.state.update(cache, md5)` here, so that things like `dvc status` don't re-calculate checksums for it. Could you also add a test for that, please? Also the comment for this line is kinda hard to wrap your head around, maybe we could improve it?

File: tests/test_add.py [hunk 2]
```diff
@@ -214,3 +219,32 @@ def test_dir(self):
...
         self.assertEqual(ret, 0)
+
+
+class TestsShouldUpdateStateEntryAfterAdd(TestDvc):
+    def test(self):
+        ret = main(["config", "cache.type", "copy"])
+        self.assertEqual(ret, 0)
+        import time
+
+        time.sleep(1)
```
Comment (efiop) [inline tests/test_add.py]: Why is this needed though?

Comment (pared) [reply tests/test_add.py]: It is not, sorry about that.

File: tests/test_add.py [hunk 3]
```diff
@@ -214,3 +220,24 @@ def test_dir(self):
...
+            ret = main(["config", "cache.type", "copy"])
+            self.assertEqual(ret, 0)
+
+            ret = main(["add", self.FOO])
+            self.assertEqual(ret, 0)
+            self.assertEqual(file_md5_counter.mock.call_count, 1)
+
+            ret = main(["status"])
+            self.assertEqual(ret, 0)
+            self.assertEqual(file_md5_counter.mock.call_count, 1)
```
Comment (efiop) [inline tests/test_add.py]: This should actually be 
```
self.assertEqual(file_md5_counter.mock.call_count, 0)
```
see https://github.com/iterative/dvc/pull/1603/files#r255385913 .
